components/automate-deployment/pkg/preflight/checks.go

```diff
@@ -364,168 +369,59 @@ func ConnectivityCheck(urls []string) Check {
-	},
-	{
-		Description: "hab-sup gossip",
-		Port:        9638,
-	},
-	{
-		Description: "pg-sidecar-service",
-		Port:        10100,
-	},
-	{
-		Description: "authn-service",
-		Port:        10113,
-	},
-	{
-		Description: "UNUSED",
-		Port:        10114,
-	},
-	{
-		Description: "session-service",
-		Port:        10115,
-	},
-	{
-		Description: "dex",
-		Port:        10116,
-	},
-	{
-		Description: "dex",
-		Port:        10117,
-	},
-	{
-		Description: "UNUSED",
-		Port:        10118,
-	},
-	{
-		Description: "config-mgmt-service",
-		Port:        10119,
-	},
-	{
-		Description: "nodemanager-service",
-		Port:        10120,
-	},
-	{
-		Description: "compliance-service",
-		Port:        10121,
-	},
-	{
-		Description: "ingest-service",
-		Port:        10122,
-	},
-	{
-		Description: "es-sidecar-service",
-		Port:        10123,
-	},
-	{
-		Description: "license-control-service",
-		Port:        10124,
-	},
-	{
-		Description: "notifications-service",
-		Port:        10125,
-	},
-	{
-		Description: "local-user-service",
-		Port:        10127,
-	},
-	{
-		Description: "teams-service",
-		Port:        10128,
-	},
-	{
-		Description: "secrets-service",
-		Port:        10131,
-	},
-	{
-		Description: "applications-service",
-		Port:        10133,
-	},
-	// TODO @afiune Uncomment once this is not a feature flag
-	//{
-	//Description: "event-service NATS",
-	//Port:        10140,
-	//},
-	// @afiune Why is the event-service not here?
-	// TODO: Add it to the checks!
-	{
-		Description: "elasticsearch",
-		Port:        10141,
-	},
-	{
-		Description: "elasticsearch",
-		Port:        10142,
-	},
-	{
-		Description: "UNUSED",
-		Port:        10159,
-	},
-	{
-		Description: "deployment-service",
-		Port:        10160,
-	},
-	{
-		Description: "automate-ui",
-		Port:        10161,
-	},
-	{
-		Description: "authn-service",
-		Port:        10162,
-	},
-	{
-		Description: "automate-load-balancer",
-		Port:        80,
-		Shared:      true,
-	},
-	{
-		Description: "automate-load-balancer",
-		Port:        443,
-		Shared:      true,
-	},
-	{
-		Description: "compliance-service",
-		Port:        2133,
-		Shared:      true,
-	},
-	{
-		Description: "compliance-service",
-		Port:        2134,
-		Shared:      true,
-	},
-	{
-		Description: "postgresql",
-		Port:        5432,
-		Shared:      true,
-	},
+func requiredPortsForConfig(skipShared bool, config *dc.AutomateConfig) []int {
+	if config == nil {
+		config = dc.DefaultAutomateConfig()
+	}
+
+	servicesToCheck, _ := deployment.ExpectedServiceIDsForConfig(config.GetDeployment()) // nolint:errcheck
+	servicesToCheck = append(servicesToCheck, habpkg.New("chef", "deployment-service"))
+
+	portsToCheck := make([]int, 0, len(servicesToCheck)+8)
+	// Append habitat ports, currently hard coded nearly everywhere.
+	portsToCheck = append(portsToCheck, 9631, 9638)
+	acValue := reflect.ValueOf(*config)
+	for i := 0; i < acValue.NumField(); i++ {
+		if v, ok := acValue.Field(i).Interface().(a2conf.A2ServiceConfig); ok {
```

Comment:
I'm thinking that instead of using reflect here, we might want to add some generated code that returns []a2conf.A2ServiceConfig